### PR TITLE
dev-vcs/git update git-2.37.2-unsafe-directory.patch

### PR DESCRIPTION
**Pull request by stefan11111**

### CODE REVIEW

File: dev-vcs/git/files/git-2.46.2-unsafe-directory.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+diff --git a/setup.c b/setup.c
```
Comment (thesamesam) [inline dev-vcs/git/files/git-2.46.2-unsafe-directory.patch:1]: You've lost the comment from the original patch.

File: dev-vcs/git/files/git-2.46.2-unsafe-directory.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+index 5f81d9f..310e34a 100644
+--- a/setup.c
++++ b/setup.c
+@@ -1305,7 +1305,7 @@ static int ensure_valid_ownership(const char *gitfile,
+ 	 * if it cannot be normalized.
+ 	 */
+ 	data.path = real_pathdup(worktree ? worktree : gitdir, 0);
+-	if (!data.path)
++	if (!data.path && git_env_bool("GIT_TEST_ASSUME_DIFFERENT_OWNER", 0))
```
Comment (thesamesam) [inline dev-vcs/git/files/git-2.46.2-unsafe-directory.patch:10]: I don't see the need for this hunk.

**Review by thesamesam — CHANGES_REQUESTED** [review]
Signed-off-by is also missing and no `Closes:` tag for the bug.

**Comment by thesamesam** [comment]
I'll fix it.

File: dev-vcs/git/files/git-2.46.2-unsafe-directory.patch (hunk 2 above)
Comment (stefan11111) [reply dev-vcs/git/files/git-2.46.2-unsafe-directory.patch:10]: This is the comment above this hunk:
``` 
   /*
     * normalize the data.path for comparison with normalized paths
     * that come from the configuration file.  The path is unsafe
     * if it cannot be normalized.
     */
```
As I read this, the above check is part of the safe directory check.

**Pull request closed without merging**